packages/jwt-middleware/README.md

```diff
@@ -0,0 +1,31 @@
+# JSON Web Token Middleware
+
+This middleware allows to secure execution of a webtask using JWT tokens by validating issuer, audience, and scope claims.
+
+## Usage
+
+1. Set the `wt-node-dependencies` metadata property to the stringified JSON of an object with names of modules as the keys and values set to the latest version for the corresponding module.
+
+```
+{
+    "@webtask/middleware-compiler": "1.3.0", 
+    "express-jwt": "5.3.1",
+    "jws": "3.1.4",
+    "jwks-rsa": "1.2.1"
+}
+```
+
+2. Set the `wt-compiler` metadata property on your webtask to `@webtask/middleware-compiler`.
+
+3. Set the `wt-compiler` metadata property to `@webtask/jwt-middleware`.
+
+4. Set the `wt-authorize-execution` metadata property to any value other than `0` to secure the webtask.  Value of `0` or missing `wt-authorize-execution` metadata property will result with the webtask execution being unsecure.
+
+5. Set the `iss` metadata property to the value of `authorization_server` property obtained from the discovery endpoint of your deployment (located at `{deployment_url}//api/description`).
+
+6. Set the `aud` metadata property to the value of `audience` property obtained from the discovery endpoint of your deployment (located at `{deployment_url}//api/description`).
```

Comment:
Are we going to default iss and aud to the ones set on the deployment *at runtime*, or do they need to be explicitly set?

Reply:
Also could we use wt-iss and wt-aud to follow the pattern of naming everything we own with `wt-*`?

Reply:
I believe they need to be explicitly set, but I might need to defer this to @ggoodman.

I will change the names to `wt-*`

Reply:
We could flow this from proxy to sandbox using x_wt header with the rest of "request metadata" so that the middleware can default to those values if not explicitly specified. The advantage of this is that a deployed Extend cluster can change the configured iss and aud over time, and all webtasks that have been created in the past will automatically inherit these settings without having to be re-created. @ggoodman thoughts?

Reply:
I think they need to be explicitly set, but will be set by the tooling cli/editor to the default values from the deployment. However, folks could override them to use whatever issuer or audience they wanted--they don't have to use the same auth server as the management APIs

Reply:
I'm on the fence here. In the spirit of 'not having anything to support until the feature is asked for', I think we should rely on clients, as @rwtombaugh suggested, to provide intelligent defaults.

Reply:
The disadvantage of requiring this is that if and when the owner of the deployment chooses to change their issuer and/or audience, all those webtasks will need to be re-created.